aiida/tools/dbimporters/plugins/materialsproject.py

```diff
@@ -45,13 +41,11 @@ def setup_db(self, **kwargs):
         except KeyError:
             try:
                 api_key = os.environ['PMG_MAPI_KEY']
-            except KeyError as exc:
-                raise_from(
-                    KeyError(
-                        'API key not supplied and PMG_MAPI_KEY environment '
-                        'variable not set. Either pass it when initializing the class, '
-                        'or set the environment variable PMG_MAPI_KEY to your API key.'
-                    ), exc
+            except KeyError:
+                KeyError(
```

Comment:
This should probably be 
```
except KeyError as exc:
    raise KeyError(...) from exc
```

That way you get the proper "error chaining" (``The above exception was the direct cause of the following exception:`` instead of ``During handling of the above exception, another exception occurred:``).

Reply:
Done